osu.Game/Beatmaps/BeatmapPanelBackgroundTextureLoaderStore.cs

```diff
@@ -66,14 +65,19 @@ private TextureUpload limitTextureUploadSize(TextureUpload textureUpload)
             textureUpload.Dispose();
 
             Size size = image.Size();
-            int usableWidth = Math.Min(max_width, size.Width);
-            int usableHeight = Math.Min(max_height, size.Height);
+
+            // Assume that panel backgrounds are always displayed using `FillMode.Fill`.
+            // Also assume that all backgrounds are wider than they are tall, so the
+            // fill is always going to be based on width.
+            //
+            // We need to include enough height to make this work for all ratio panels are displayed at.
+            int usableHeight = (int)Math.Ceiling(size.Width * 1 / minimum_display_ratio);
```

Comment:
Not sure how much we care, but if the image is shorter than `usableHeight` (which can happen, i.e. say if you try to resize a 1000x10 image to fit the panel), this will fail because i# can't crop to bounds bigger than the source image.

Can be fixed via something like

```diff
diff --git a/osu.Game/Beatmaps/BeatmapPanelBackgroundTextureLoaderStore.cs b/osu.Game/Beatmaps/BeatmapPanelBackgroundTextureLoaderStore.cs
index 110cebbe0e..dcef1fff21 100644
--- a/osu.Game/Beatmaps/BeatmapPanelBackgroundTextureLoaderStore.cs
+++ b/osu.Game/Beatmaps/BeatmapPanelBackgroundTextureLoaderStore.cs
@@ -72,6 +72,7 @@ private TextureUpload limitTextureUploadSize(TextureUpload textureUpload)
             //
             // We need to include enough height to make this work for all ratio panels are displayed at.
             int usableHeight = (int)Math.Ceiling(size.Width * 1 / minimum_display_ratio);
+            usableHeight = Math.Min(usableHeight, size.Height);
 
             // Crop the centre region of the background for now.
             Rectangle cropRectangle = new Rectangle(
```

I'm not sure if it will visually look the same, but I don't really care. I care about not having this edge case throw errors, and only that.